diff --git include/math/aabb.hpp include/math/aabb.hpp
index 6e5114e..1824756 100644
--- include/math/aabb.hpp
+++ include/math/aabb.hpp
@@ -15,10 +15,6 @@ namespace rive
 			float buffer[4];
 			struct
 			{
-				Vec2D min, max;
-			};
-			struct
-			{
 				float minX, minY, maxX, maxY;
 			};
 		};
@@ -48,4 +44,4 @@ namespace rive
 		float perimeter() const;
 	};
 } // namespace rive
-#endif
\ No newline at end of file
+#endif
diff --git src/math/aabb.cpp src/math/aabb.cpp
index 94de5d0..a7ea1db 100644
--- src/math/aabb.cpp
+++ src/math/aabb.cpp
@@ -77,9 +77,9 @@ bool AABB::areIdentical(const AABB& a, const AABB& b)
 	return a[0] == b[0] && a[1] == b[1] && a[2] == b[2] && a[3] == b[3];
 }
 
-float AABB::width() const { return max[0] - min[0]; }
+float AABB::width() const { return buffer[2] - buffer[0]; }
 
-float AABB::height() const { return max[1] - min[1]; }
+float AABB::height() const { return buffer[3] - buffer[1]; }
 
 float AABB::perimeter() const
 {
